### PR TITLE
Update leak() to use this.ptr

### LINKED ISSUE #284: DiplomatBuf finalizer undefined variable
On this line:

```javascript
        console.error(`Failed to unregister DiplomatBuf at ${ptr}, this is a bug. Either it was never registered (leak), it was already unregistered (failed attempt to double free), or the unregister token was unrecognized (fallback to GC).`);
```

the variable `ptr` is not declared. There is a similar console log a few lines above that is fine.

### PR DESCRIPTION
**Pull request by Manishearth**
fixes #284

### CODE REVIEW

File: example/js/lib/api/diplomat-runtime.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ export class DiplomatBuf {
       if (successfully_unregistered) {
         // leak
       } else {
-        console.error(`Failed to unregister DiplomatBuf at ${ptr}, this is a bug. Either it was never registered (leak), it was already unregistered (failed attempt to double free), or the unregister token was unrecognized (fallback to GC).`);
+        console.error(`Failed to unregister DiplomatBuf at ${this.ptr}, this is a bug. Either it was never registered (leak), it was already unregistered (failed attempt to double free), or the unregister token was unrecognized (fallback to GC).`);
```
Comment (sffc) [inline example/js/lib/api/diplomat-runtime.js:114]: Question: why is `leak` a member function but `free` is declared inline?

**Review by sffc — APPROVED** [review]

File: example/js/lib/api/diplomat-runtime.js (hunk 1 above)
Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:114]: no idea!

Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:114]: i think they were written in different PRs and I didn't notice the inconsistency

Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:114]: Merging this fix but I can look into making this consistent later

Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:114]: Oh it's because free() needs additional ctor-time context, leak doesn't.

Comment (sffc) [reply example/js/lib/api/diplomat-runtime.js:114]: It only seems to be `align` which seems like something that should probably be stored on the instance anyway

**Pull request merged**